adlfs/spec.py

```diff
@@ -862,32 +870,115 @@ async def _glob(self, path, **kwargs):
+
+                # Check the depth that needs to be screened
+                depth = target_path.count("/")
+                outblobs = []
+                try:
+                    async for next_blob in blobs:
+                        if depth in [0, 1] and path == "":
+                            outblobs.append(next_blob)
+                        elif isinstance(next_blob, BlobProperties):
+                            if next_blob["name"].count("/") == depth:
+                                outblobs.append(next_blob)
+                            elif not next_blob["name"].endswith("/") and (
+                                next_blob["name"].count("/") == (depth - 1)
+                            ):
+                                outblobs.append(next_blob)
+                        else:
+                            async for blob_ in next_blob:
+                                if isinstance(blob_, BlobProperties) or isinstance(
+                                    blob_, BlobPrefix
+                                ):
+                                    if blob_["name"].endswith("/"):
+                                        if (
+                                            blob_["name"].rstrip("/").count("/")
+                                            == depth
+                                        ):
+                                            outblobs.append(blob_)
+                                        elif blob_["name"].count("/") == depth and (
+                                            hasattr(blob_, "size")
+                                            and blob_["size"] == 0
+                                        ):
+                                            outblobs.append(blob_)
+                                        else:
+                                            pass
+                                    elif blob_["name"].count("/") == (depth):
+                                        outblobs.append(blob_)
+                                    else:
+                                        pass
+                except ResourceNotFoundError:
+                    raise FileNotFoundError
+                finalblobs = await self._details(
+                    outblobs,
+                    target_path=target_path,
+                    return_glob=return_glob,
+                    version_id=version_id,
+                )
+                if return_glob:
+                    return finalblobs
+                finalblobs = await self._details(
+                    outblobs, target_path=target_path, version_id=version_id
+                )
+                if not finalblobs:
+                    if not await self._exists(target_path):
+                        raise FileNotFoundError
+                    return []
+                if not self.version_aware or finalblobs[0].get("is_current_version"):
+                    self.dircache[target_path] = finalblobs
+                return finalblobs
```

Comment:
This got changed from
```
if self.version_aware or not finalblobs[0].get("is_current_version"):
    return finalblobs
cache[target_path] = finalblobs
self.dircache[target_path] = finalblobs
```

to make it more readable and closer to the way such operations are done elsewhere.